addons/account/models/account_move.py

```diff
@@ -1859,8 +1839,13 @@ def post(self):
             if not move.invoice_date and move.is_invoice(include_receipts=True):
                 move.invoice_date = fields.Date.context_today(self)
                 move.with_context(check_move_validity=False)._onchange_invoice_date()
+            if move.company_id.auto_correct_accounting_date \
+                    and move.date <= (move.company_id.tax_lock_date or date.min) \
+                    and any(line.tax_ids or line.tax_line_id for line in move.line_ids):
+                move.date = move.company_id.tax_lock_date
```

Comment:
It should be `move.company_id.tax_lock_date + one day` as you compare dates with `<=`